Sources/AsyncHTTPClient/RequestBag.swift

```diff
@@ -154,8 +154,16 @@ final class RequestBag<Delegate: HTTPClientResponseDelegate> {
             return self.task.eventLoop.makeFailedFuture(error)
 
         case .write(let part, let writer, let future):
-            writer.writeRequestBodyPart(part, request: self)
-            self.delegate.didSendRequestPart(task: self.task, part)
+            let promise = self.task.eventLoop.makePromise(of: Void.self)
+            promise.futureResult.whenComplete { result in
+                switch result {
+                case .success:
+                    self.delegate.didSendRequestPart(task: self.task, part)
+                case .failure(let error):
+                    self.delegate.didFailSendingRequestPart(task: self.task, part, error: error)
```

Comment:
I think, we should not change the old API at the moment and should just ignore this case. The error will be reported to the user anyway.